sample/src/sample.ts

```diff
@@ -3,9 +3,19 @@ import { Server as HttpServer } from "http";
 
 export function attachWebSockets(server: HttpServer) {
   runWebsocketServer({
-    onAuthenticate: async function(msg: any) {
-      console.log("onAuthenticate: " + JSON.stringify(msg));
-      return "success";
+    auth: {
+      canRead: ({ auth, kind, id }) => {
+        console.log("canRead", auth, kind, id);
```

Comment:
I am curious to know your thoughts around finer grained authorization. By having Oncilla implement very coarse authz, if the server ever needs more fine grained authorization (e.g. can read type a but not type b), there will be a separation of concerns in this implementation.

Reply:
I don’t understand. Oncilla would not make any decisions, it exposes this function here exactly so that we could do individual things per the needs of our domain app:

```ts
canRead: ({ auth, kind, id }) => {
  if (auth.isAdmin) return true;
  if (kind === 'tasks') return true;
  return false;
}
```